src/client/Microsoft.Identity.Client/ManagedIdentity/ImdsManagedIdentitySource.cs

```diff
@@ -56,14 +56,16 @@ internal ImdsManagedIdentitySource(RequestContext requestContext) :
             requestContext.Logger.Verbose(() => "[Managed Identity] Creating IMDS managed identity source. Endpoint URI: " + _imdsEndpoint);
         }
 
-        protected override ManagedIdentityRequest CreateRequest(string resource)
+        protected override ManagedIdentityRequest CreateRequest(string resource, AcquireTokenForManagedIdentityParameters parameters)
         {
             ManagedIdentityRequest request = new(HttpMethod.Get, _imdsEndpoint);
 
             request.Headers.Add("Metadata", "true");
             request.QueryParameters["api-version"] = ImdsApiVersion;
             request.QueryParameters["resource"] = resource;
 
+            ApplyClaimsAndCapabilities(request, parameters);
```

Comment:
Isn't IMDS going to MSIv2?

Reply:
VM/VMSS will go to MSI v2. IMDS as a source will be supported by other MIRPs

Reply:
Will remove MSI v1 Token Revocation from IMDS code path. As ACI is the only one that will use the code path, and they do not plan to support Token Revocation yet.